### ISSUE TITLE
Firefox 47 crashes upon start

### ISSUE THREAD

**Issue by randomshinichi** [SeleniumHQ/selenium#2579]
## Meta -
OS:  
OS X 10.11
Selenium Version:  
2.53.1-2.53.6
Browser:  
Firefox

-->

Browser Version:  
47.0 64bit

## Expected Behavior -
Firefox opens
## Actual Behavior -
Firefox exits before selenium can connect to it. Selenium reports
```
Traceback (most recent call last):
File "functional_tests.py", line 2, in
browser = webdriver.Firefox()
File "/home/peter/Documents/Virtenvs/all/b2bapp/lib/python3.5/site-packages/selenium/webdriver/firefox/webdriver.py", line 80, in init
self.binary, timeout)
File "/home/peter/Documents/Virtenvs/all/b2bapp/lib/python3.5/site-packages/selenium/webdriver/firefox/extension_connection.py", line 52, in init
self.binary.launch_browser(self.profile, timeout=timeout)
File "/home/peter/Documents/Virtenvs/all/b2bapp/lib/python3.5/site-packages/selenium/webdriver/firefox/firefox_binary.py", line 68, in launch_browser
self._wait_until_connectable(timeout=timeout)
File "/home/peter/Documents/Virtenvs/all/b2bapp/lib/python3.5/site-packages/selenium/webdriver/firefox/firefox_binary.py", line 99, in _wait_until_connectable
"The browser appears to have exited "
selenium.common.exceptions.WebDriverException: Message: The browser appears to have exited before we could connect. If you specified a log_file in the FirefoxBinary constructor, check it for details.
```

log_file:
```
1470447997417   addons.manager  DEBUG   Application has been upgraded
1470447997440   addons.manager  DEBUG   Loaded provider scope for resource://gre/modules/addons/XPIProvider.jsm: ["XPIProvider"]
1470447997442   addons.manager  DEBUG   Loaded provider scope for resource://gre/modules/LightweightThemeManager.jsm: ["LightweightThemeManager"]
1470447997447   addons.manager  DEBUG   Loaded provider scope for resource://gre/modules/addons/GMPProvider.jsm
1470447997449   addons.manager  DEBUG   Loaded provider scope for resource://gre/modules/addons/PluginProvider.jsm
1470447997450   addons.manager  DEBUG   Starting provider: XPIProvider
1470447997451   addons.xpi  DEBUG   startup
1470447997451   addons.xpi  INFO    Mapping fxdriver@googlecode.com to /var/folders/_z/tg84rj7d38d093h0xvpn3v9r0000gn/T/tmpjgu1hxcu/extensions/fxdriver@googlecode.com
1470447997452   addons.xpi  INFO    SystemAddonInstallLocation directory is missing
1470447997453   addons.xpi  INFO    Mapping e10srollout@mozilla.org to /Applications/Firefox.app/Contents/Resources/browser/features/e10srollout@mozilla.org.xpi
1470447997453   addons.xpi  INFO    Mapping firefox@getpocket.com to /Applications/Firefox.app/Contents/Resources/browser/features/firefox@getpocket.com.xpi
1470447997453   addons.xpi  INFO    Mapping loop@mozilla.org to /Applications/Firefox.app/Contents/Resources/browser/features/loop@mozilla.org.xpi
1470447997454   addons.xpi  INFO    Mapping {972ce4c6-7e08-4474-a285-3208198ce6fd} to /Applications/Firefox.app/Contents/Resources/browser/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}.xpi
1470447997454   addons.xpi  DEBUG   Skipping unavailable install location app-system-share
1470447997455   addons.xpi  INFO    Mapping web2pdfextension@web2pdf.adobedotcom to /Library/Application Support/Mozilla/Extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}/web2pdfextension@web2pdf.adobedotcom
1470447997456   addons.xpi  DEBUG   checkForChanges
1470447997457   addons.xpi  DEBUG   Loaded add-on state from prefs: {}
1470447997457   addons.xpi  DEBUG   New add-on fxdriver@googlecode.com in app-profile
1470447997458   addons.xpi  DEBUG   getModTime: Recursive scan of fxdriver@googlecode.com
1470447997461   addons.xpi  DEBUG   New add-on e10srollout@mozilla.org in app-system-defaults
1470447997461   addons.xpi  DEBUG   getModTime: Recursive scan of e10srollout@mozilla.org
1470447997462   addons.xpi  DEBUG   New add-on firefox@getpocket.com in app-system-defaults
1470447997462   addons.xpi  DEBUG   getModTime: Recursive scan of firefox@getpocket.com
1470447997462   addons.xpi  DEBUG   New add-on loop@mozilla.org in app-system-defaults
1470447997462   addons.xpi  DEBUG   getModTime: Recursive scan of loop@mozilla.org
1470447997463   addons.xpi  DEBUG   New add-on {972ce4c6-7e08-4474-a285-3208198ce6fd} in app-global
1470447997463   addons.xpi  DEBUG   getModTime: Recursive scan of {972ce4c6-7e08-4474-a285-3208198ce6fd}
1470447997464   addons.xpi  DEBUG   New add-on web2pdfextension@web2pdf.adobedotcom in app-system-local
1470447997464   addons.xpi  DEBUG   getModTime: Recursive scan of web2pdfextension@web2pdf.adobedotcom
1470447997465   addons.xpi  DEBUG   getInstallState changed: true, state: {"app-profile":{"fxdriver@googlecode.com":{"d":"/var/folders/_z/tg84rj7d38d093h0xvpn3v9r0000gn/T/tmpjgu1hxcu/extensions/fxdriver@googlecode.com","st":1470447997000,"mt":1470447997000}},"app-system-defaults":{"e10srollout@mozilla.org":{"d":"/Applications/Firefox.app/Contents/Resources/browser/features/e10srollout@mozilla.org.xpi","st":1465554665000},"firefox@getpocket.com":{"d":"/Applications/Firefox.app/Contents/Resources/browser/features/firefox@getpocket.com.xpi","st":1465554665000},"loop@mozilla.org":{"d":"/Applications/Firefox.app/Contents/Resources/browser/features/loop@mozilla.org.xpi","st":1465554665000}},"app-global":{"{972ce4c6-7e08-4474-a285-3208198ce6fd}":{"d":"/Applications/Firefox.app/Contents/Resources/browser/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}.xpi","st":1465554665000}},"app-system-local":{"web2pdfextension@web2pdf.adobedotcom":{"d":"/Library/Application Support/Mozilla/Extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}/web2pdfextension@web2pdf.adobedotcom","st":1469108621000,"mt":1466710987000}}}
1470447997474   addons.xpi-utils    DEBUG   Opening XPI database /var/folders/_z/tg84rj7d38d093h0xvpn3v9r0000gn/T/tmpjgu1hxcu/extensions.json
1470447997476   addons.xpi-utils    DEBUG   New add-on fxdriver@googlecode.com installed in app-profile
*** Blocklist::_loadBlocklistFromFile: blocklist is disabled
1470447997493   DeferredSave.extensions.json    DEBUG   Save changes
1470447997494   addons.xpi-utils    DEBUG   New add-on e10srollout@mozilla.org installed in app-system-defaults
1470447997497   DeferredSave.extensions.json    DEBUG   Starting timer
1470447997499   DeferredSave.extensions.json    DEBUG   Save changes
1470447997499   addons.xpi-utils    DEBUG   New add-on firefox@getpocket.com installed in app-system-defaults
1470447997505   DeferredSave.extensions.json    DEBUG   Save changes
1470447997506   addons.xpi-utils    DEBUG   New add-on loop@mozilla.org installed in app-system-defaults
1470447997515   DeferredSave.extensions.json    DEBUG   Save changes
1470447997515   addons.xpi-utils    DEBUG   New add-on {972ce4c6-7e08-4474-a285-3208198ce6fd} installed in app-global
1470447997518   DeferredSave.extensions.json    DEBUG   Save changes
1470447997519   addons.xpi-utils    DEBUG   New add-on web2pdfextension@web2pdf.adobedotcom installed in app-system-local
1470447997527   DeferredSave.extensions.json    DEBUG   Starting write
1470447997536   addons.xpi-utils    WARN    Disabling foreign installed add-on web2pdfextension@web2pdf.adobedotcom in app-system-local
1470447997536   DeferredSave.extensions.json    DEBUG   Save changes
1470447997536   DeferredSave.extensions.json    DEBUG   Data changed while write in progress
1470447997537   addons.xpi-utils    INFO    Hiding the updated system add-ons.
1470447997537   addons.manager  DEBUG   Registering startup change 'installed' for fxdriver@googlecode.com
1470447997537   addons.xpi-utils    DEBUG   Make addon app-profile:fxdriver@googlecode.com visible
1470447997537   DeferredSave.extensions.json    DEBUG   Save changes
1470447997545   addons.xpi  DEBUG   Loading bootstrap scope from /Applications/Firefox.app/Contents/Resources/browser/features/e10srollout@mozilla.org.xpi
1470447997559   addons.xpi  DEBUG   Calling bootstrap method install on e10srollout@mozilla.org version 1.0
1470447997560   addons.xpi-utils    DEBUG   Make addon app-system-defaults:e10srollout@mozilla.org visible
1470447997560   DeferredSave.extensions.json    DEBUG   Save changes
1470447997561   addons.xpi  DEBUG   Loading bootstrap scope from /Applications/Firefox.app/Contents/Resources/browser/features/firefox@getpocket.com.xpi
1470447997567   addons.xpi  DEBUG   Calling bootstrap method install on firefox@getpocket.com version 1.0.2
1470447997567   addons.xpi-utils    DEBUG   Make addon app-system-defaults:firefox@getpocket.com visible
1470447997567   DeferredSave.extensions.json    DEBUG   Save changes
1470447997568   addons.xpi  DEBUG   Loading bootstrap scope from /Applications/Firefox.app/Contents/Resources/browser/features/loop@mozilla.org.xpi
1470447997575   addons.xpi  DEBUG   Calling bootstrap method install on loop@mozilla.org version 1.3.2
1470447997575   addons.xpi-utils    DEBUG   Make addon app-system-defaults:loop@mozilla.org visible
1470447997576   DeferredSave.extensions.json    DEBUG   Save changes
1470447997576   addons.xpi-utils    DEBUG   Make addon app-global:{972ce4c6-7e08-4474-a285-3208198ce6fd} visible
1470447997577   DeferredSave.extensions.json    DEBUG   Save changes
1470447997577   addons.manager  DEBUG   Registering startup change 'installed' for web2pdfextension@web2pdf.adobedotcom
1470447997577   addons.xpi-utils    DEBUG   Make addon app-system-local:web2pdfextension@web2pdf.adobedotcom visible
1470447997577   DeferredSave.extensions.json    DEBUG   Save changes
1470447997578   addons.xpi  DEBUG   Updating XPIState for {"id":"fxdriver@googlecode.com","syncGUID":"Mg92LaRrtL18","location":"app-profile","version":"2.53.0","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Firefox WebDriver","description":"WebDriver implementation for Firefox","creator":"Simon Stewart","homepageURL":null},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"/var/folders/_z/tg84rj7d38d093h0xvpn3v9r0000gn/T/tmpjgu1hxcu/extensions/fxdriver@googlecode.com","installDate":1470447997000,"updateDate":1470447997000,"applyBackgroundUpdates":1,"bootstrap":false,"skinnable":false,"size":3249525,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":true,"hasBinaryComponents":true,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"3.0","maxVersion":"48.0"}],"targetPlatforms":[{"os":"Darwin","abi":null},{"os":"SunOS","abi":null},{"os":"FreeBSD","abi":null},{"os":"OpenBSD","abi":null},{"os":"WINNT","abi":null},{"os":"Linux","abi":null}],"multiprocessCompatible":false,"signedState":0,"seen":true}
1470447997578   addons.xpi  DEBUG   getModTime: Recursive scan of fxdriver@googlecode.com
1470447997581   addons.xpi  DEBUG   Updating XPIState for {"id":"e10srollout@mozilla.org","syncGUID":"KH3y_vJ2zzYG","location":"app-system-defaults","version":"1.0","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Multi-process staged rollout","description":"Staged rollout of Firefox multi-process feature.","creator":null,"homepageURL":null},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"/Applications/Firefox.app/Contents/Resources/browser/features/e10srollout@mozilla.org.xpi","installDate":1465554665000,"updateDate":1465554665000,"applyBackgroundUpdates":1,"bootstrap":true,"skinnable":false,"size":4580,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"47.0","maxVersion":"47.0"}],"targetPlatforms":[],"multiprocessCompatible":false,"seen":true}
1470447997581   addons.xpi  DEBUG   getModTime: Recursive scan of e10srollout@mozilla.org
1470447997581   addons.xpi  DEBUG   Updating XPIState for {"id":"firefox@getpocket.com","syncGUID":"vqkdWdGOkfZK","location":"app-system-defaults","version":"1.0.2","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Pocket","description":"When you find something you want to view later, put it in Pocket.","creator":null,"homepageURL":null},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"/Applications/Firefox.app/Contents/Resources/browser/features/firefox@getpocket.com.xpi","installDate":1465554665000,"updateDate":1465554665000,"applyBackgroundUpdates":1,"bootstrap":true,"skinnable":false,"size":894473,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"47.0","maxVersion":"47.0"}],"targetPlatforms":[],"multiprocessCompatible":false,"seen":true}
1470447997581   addons.xpi  DEBUG   getModTime: Recursive scan of firefox@getpocket.com
1470447997582   addons.xpi  DEBUG   Updating XPIState for {"id":"loop@mozilla.org","syncGUID":"itc8lrnnKNGI","location":"app-system-defaults","version":"1.3.2","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Firefox Hello","description":"Web sharing for Firefox","creator":"Mozilla","homepageURL":null},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"/Applications/Firefox.app/Contents/Resources/browser/features/loop@mozilla.org.xpi","installDate":1465554665000,"updateDate":1465554665000,"applyBackgroundUpdates":1,"bootstrap":true,"skinnable":false,"size":4239140,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"46.0a1","maxVersion":"47.*"},{"id":"xpcshell@tests.mozilla.org","minVersion":"0","maxVersion":"10"}],"targetPlatforms":[],"multiprocessCompatible":false,"seen":true}
1470447997582   addons.xpi  DEBUG   getModTime: Recursive scan of loop@mozilla.org
1470447997582   addons.xpi  DEBUG   Updating XPIState for {"id":"{972ce4c6-7e08-4474-a285-3208198ce6fd}","syncGUID":"TcqbUrxR2Dro","location":"app-global","version":"47.0","type":"theme","internalName":"classic/1.0","updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{"32":"icon.png","48":"icon.png"},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Default","description":"The default theme.","creator":"Mozilla","homepageURL":null,"contributors":["Mozilla Contributors"]},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"/Applications/Firefox.app/Contents/Resources/browser/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}.xpi","installDate":1465554665000,"updateDate":1465554665000,"applyBackgroundUpdates":1,"skinnable":true,"size":7134,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":true,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"47.0","maxVersion":"47.0"}],"targetPlatforms":[],"seen":true}
1470447997583   addons.xpi  DEBUG   getModTime: Recursive scan of {972ce4c6-7e08-4474-a285-3208198ce6fd}
1470447997583   addons.xpi  DEBUG   Updating XPIState for {"id":"web2pdfextension@web2pdf.adobedotcom","syncGUID":"FPrND2wO6yoG","location":"app-system-local","version":"2.0","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":"chrome://web2pdfextension/skin/C_AdobeLogo_48x48.png","icon64URL":"chrome://web2pdfextension/skin/C_AdobeLogo_64x64.png","defaultLocale":{"name":"Adobe Acrobat - Create PDF","description":"Converts a web page to PDF","creator":"Adobe Systems Incorporated","homepageURL":"http://www.adobe.com/"},"visible":true,"active":false,"userDisabled":true,"appDisabled":false,"descriptor":"/Library/Application Support/Mozilla/Extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}/web2pdfextension@web2pdf.adobedotcom","installDate":1469108621000,"updateDate":1469108621000,"applyBackgroundUpdates":1,"bootstrap":false,"skinnable":false,"size":148797,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":true,"hasBinaryComponents":false,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"4.0","maxVersion":"46.0"}],"targetPlatforms":[],"multiprocessCompatible":false,"signedState":2,"seen":true}
1470447997584   DeferredSave.extensions.json    DEBUG   Save changes
1470447997584   addons.xpi  DEBUG   Updating database with changes to installed add-ons
1470447997584   addons.xpi-utils    DEBUG   Updating add-on states
1470447997585   addons.xpi-utils    DEBUG   Writing add-ons list
1470447997586   addons.xpi  DEBUG   Registering manifest for /Applications/Firefox.app/Contents/Resources/browser/features/e10srollout@mozilla.org.xpi
1470447997587   addons.xpi  DEBUG   Calling bootstrap method startup on e10srollout@mozilla.org version 1.0
1470447997587   addons.xpi  DEBUG   Registering manifest for /Applications/Firefox.app/Contents/Resources/browser/features/firefox@getpocket.com.xpi
1470447997587   addons.xpi  DEBUG   Calling bootstrap method startup on firefox@getpocket.com version 1.0.2
1470447997588   addons.xpi  DEBUG   Registering manifest for /Applications/Firefox.app/Contents/Resources/browser/features/loop@mozilla.org.xpi
1470447997589   addons.xpi  DEBUG   Calling bootstrap method startup on loop@mozilla.org version 1.3.2
1470447997612   addons.manager  DEBUG   Registering shutdown blocker for XPIProvider
1470447997613   addons.manager  DEBUG   Provider finished startup: XPIProvider
1470447997613   addons.manager  DEBUG   Starting provider: LightweightThemeManager
1470447997613   addons.manager  DEBUG   Registering shutdown blocker for LightweightThemeManager
1470447997614   addons.manager  DEBUG   Provider finished startup: LightweightThemeManager
1470447997614   addons.manager  DEBUG   Starting provider: GMPProvider
1470447997621   addons.manager  DEBUG   Registering shutdown blocker for GMPProvider
1470447997622   addons.manager  DEBUG   Provider finished startup: GMPProvider
1470447997622   addons.manager  DEBUG   Starting provider: PluginProvider
1470447997622   addons.manager  DEBUG   Registering shutdown blocker for PluginProvider
1470447997623   addons.manager  DEBUG   Provider finished startup: PluginProvider
1470447997623   addons.manager  DEBUG   Completed startup sequence
1470447998023   addons.manager  DEBUG   Starting provider: <unnamed-provider>
1470447998023   addons.manager  DEBUG   Registering shutdown blocker for <unnamed-provider>
1470447998023   addons.manager  DEBUG   Provider finished startup: <unnamed-provider>
1470447998156   DeferredSave.extensions.json    DEBUG   Write succeeded
1470447998156   addons.xpi-utils    DEBUG   XPI Database saved, setting schema version preference to 17
1470447998156   DeferredSave.extensions.json    DEBUG   Starting timer
1470447998178   addons.repository   DEBUG   No addons.json found.
1470447998179   DeferredSave.addons.json    DEBUG   Save changes
1470447998181   DeferredSave.addons.json    DEBUG   Starting timer
1470447998222   addons.manager  DEBUG   Starting provider: PreviousExperimentProvider
1470447998222   addons.manager  DEBUG   Registering shutdown blocker for PreviousExperimentProvider
1470447998222   addons.manager  DEBUG   Provider finished startup: PreviousExperimentProvider
1470447998226   DeferredSave.extensions.json    DEBUG   Starting write
1470447998234   DeferredSave.addons.json    DEBUG   Starting write
1470447998250   DeferredSave.extensions.json    DEBUG   Write succeeded
1470447998264   DeferredSave.addons.json    DEBUG   Write succeeded
```
## Steps to reproduce -
```
from selenium import webdriver
webdriver.Firefox()
```

**Comment by rtrembleau** [comment]
Hi, 

Had the same issue but seems to be fixed on firefox 47.0.1. 
https://www.mozilla.org/en-US/firefox/47.0.1/releasenotes/

RT.

**Comment by AutomatedTester** [comment]
Upgrade Firefox for it to work
